classes/order/Order.php

```diff
@@ -2463,4 +2463,65 @@ public function getOrderDetailTaxes()
             'WHERE o.id_order = '.(int)$this->id
         );
     }
+
+    /**
+     * Re calculate shipping cost
+     * @return object $order
+     */
+    public function refreshShippingCost()
+    {
+        if (empty($this->id)) {
+            return false;
+        }
+
+        if (!Configuration::get('PS_ORDER_RECALCULATE_SHIPPING')) {
+            return $this;
+        }
+
+        $fake_cart = new Cart($this->id_cart);
+        $new_cart = $fake_cart->duplicate();
+        $new_cart = $new_cart['cart'];
+
+        // assign order id_address_delivery to cart
+        $new_cart->id_address_delivery = $this->id_address_delivery;
+
+        // assign id_carrier
+        $new_cart->id_carrier = $this->id_carrier;
+
+        //remove all products : cart (maybe change in the meantime)
+        foreach ($new_cart->getProducts() as $product) {
+            $new_cart->deleteProduct($product['id_product'], $product['id_product_attribute']);
+        }
+
+        // add real order products
+        foreach ($this->getProducts() as $product) {
+            $new_cart->updateQty($product['product_quantity'], $product['product_id']);
+        }
+
+        // get new shipping cost
+        $base_total_shipping_tax_incl = (float)$new_cart->getPackageShippingCost((int)$new_cart->id_carrier, true, null);
```

Comment:
I believe that this standard (PSR-2) may no longer be used for pull requests. Gotta use `camelCase`, spaces after casts (bleghhh) and yoda conditions (don't get me started on those haha).